cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -86,12 +97,146 @@ func (m *Manager) Deploy(ctx context.Context) (*DeployResult, error) {
 		return nil, fmt.Errorf("updating environment with deployment outputs: %w", err)
 	}
 
+	infraRoot := m.options.Path
+	if !filepath.IsAbs(infraRoot) {
+		infraRoot = filepath.Join(m.projectPath, m.options.Path)
+	}
+	fileShareUploadOperations, err := azdFileShareUploadOperations(infraRoot, *m.env)
+	if err != nil {
+		return nil, fmt.Errorf("looking for azd fileShare upload operations: %w", err)
+	}
+
+	if len(fileShareUploadOperations) > 0 {
+		m.console.ShowSpinner(ctx, "uploading files to fileShare", input.StepFailed)
+	}
+	for _, op := range fileShareUploadOperations {
+		if err := bindMountOperation(
+			ctx,
+			m.fileShareService,
+			m.cloud.StorageEndpointSuffix,
+			m.env.GetSubscriptionId(),
+			op.StorageAccount,
+			op.FileShareName,
+			op.Path); err != nil {
+			return nil, fmt.Errorf("error binding mount: %w", err)
+		}
+		m.console.MessageUxItem(ctx, &ux.DisplayedResource{
+			Type:  fileShareUploadOperation,
+			Name:  op.Description,
+			State: ux.SucceededState,
+		})
+	}
+
 	// make sure any spinner is stopped
 	m.console.StopSpinner(ctx, "", input.StepDone)
 
 	return deployResult, nil
 }
 
+const (
+	fileShareUploadOperation string = "FileShareUpload"
+	azdOperationsFileName    string = "azd.operations.yaml"
+)
+
+type azdOperation struct {
+	Type        string
+	Description string
+	Config      any
+}
+
+type azdOperationFileShareUpload struct {
+	Description    string
+	StorageAccount string
+	FileShareName  string
+	Path           string
+}
+
+type azdOperationsModel struct {
+	Operations []azdOperation
+}
+
+func azdOperations(infraPath string, env environment.Environment) (azdOperationsModel, error) {
+	// Check if the file exists
+	path := filepath.Join(infraPath, azdOperationsFileName)
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// File does not exist
+			return azdOperationsModel{}, nil
```

Comment:
NIT: You might consider just using `errors.Is` to test for os.ErrNotFound on the return value from `os.ReadFile` instead of doing the check and then then the read.

Reply:
updated it